packages/models-library/src/models_library/service_settings_labels.py

```diff
@@ -165,14 +168,75 @@ class PathMappingsLabel(BaseModel):
         description="optional list unix shell rules used to exclude files from the state",
     )
 
+    volume_size_limits: Optional[dict[str, str]] = Field(
+        None,
+        description=(
+            "Apply volume size limits to entries in: `inputs_path`, `outputs_path` "
+            "and `state_paths`. Limits must be parsable by Pydantic's ByteSize."
+        ),
+    )
+
+    @validator("volume_size_limits")
+    @classmethod
+    def validate_volume_limits(cls, v, values) -> Optional[str]:
+        if v is None:
+            return v
+
+        for path_str, size_str in v.items():
+            # checks that format is correct
+            try:
+                parse_obj_as(ByteSize, size_str)
```

Comment:
The same validation can be simply achieved simply adding it in the type annotation as

```python
volume_size_limits: Optional[dict[str, ByteSize]] 
```

NOTE that ``ByteSize`` is actually a ``str`` with class methods for validation!

Reply:
@pcrespov ByteSize is not a string, but bytes or int actually.